include/boost/url/params_encoded.hpp

```diff
@@ -49,9 +49,19 @@ class params_encoded
         url& u) noexcept;
 
 public:
-    /** A read-only random-access iterator to an encoded query parameter.
-    */
+    /** A read-only forward iterator to an encoded query parameter.
+
+        This is a read-only forward iterator to
+        the encoded query parameters. It has the
+        value type @ref query_param and reference
+        type @ref query_param_view.
+
+     */
+#ifdef BOOST_URL_DOCS
+    using iterator = __see_below__;
+#else
     class iterator;
+#endif
 
     /** A read-only random-access iterator to an encoded query parameter.
```

Comment:
well is it random access or is it forward?

Reply:
Oh...